docs/Setup_Kubernetes_Cluster.md

```diff
@@ -169,7 +169,211 @@ ubuntu                                  Ready    <none>          14s     v1.25.5
+♻️  Restarting unmanaged pods...
+♻️  Restarted unmanaged pod kube-system/coredns-565d847f94-sn4ht
+✅ Cilium was successfully installed! Run 'cilium status' to view installation health
+
+> cilium status
+    /¯¯\
+/¯¯\__/¯¯\    Cilium:         OK
+\__/¯¯\__/    Operator:       OK
+/¯¯\__/¯¯\    Hubble:         disabled
+\__/¯¯\__/    ClusterMesh:    disabled
+    \__/
+
+Deployment        cilium-operator    Desired: 1, Ready: 1/1, Available: 1/1
+DaemonSet         cilium             Desired: 2, Ready: 2/2, Available: 2/2
+Containers:       cilium             Running: 2
+                  cilium-operator    Running: 1
+Cluster Pods:     2/3 managed by Cilium
+Image versions    cilium             quay.io/cilium/cilium:v1.12.5@sha256:06ce2b0a0a472e73334a7504ee5c5d8b2e2d7b72ef728ad94e564740dd505be5: 2
+                  cilium-operator    quay.io/cilium/operator-generic:v1.12.5@sha256:b296eb7f0f7656a5cc19724f40a8a7121b7fd725278b7d61dc91fe0b7ffd7c0e: 1
+
+> kubectl get pods -A | grep cilium
+kube-system   cilium-operator-74b8595d7b-v6j42                                1/1     Running   0          3m43s
+kube-system   cilium-pbntp                                                    1/1     Running   0          3m43s
+kube-system   cilium-qxbd9                                                    1/1     Running   0          3m43s
+```
+
+#### Enable Security Encryption
+
+Cilium provides security encryption feature either IPsec or Wireguard statically during initialization.
+Packets are not encrypted when they are destined to the same node from which they were sent. This behavior is intended.
+Encryption would provide no benefits in that case, given that the raw traffic can be observed on the node anyway.
+Encryption requires key, the following procedure uses Kubernetes Secret to bind the key for cilium encryption.
+
+```bash
+### If cilium is already deployed, uninstall from cluster system
+> cilium uninstall
+
+### Create Kubernetes Secret that is used for encryption
+> PSK=($(dd if=/dev/urandom count=20 bs=1 2> /dev/null | xxd -p -c 64))
+
+> echo $PSK
+47b57a0241e4c5df0a196ccfbcc7ee1aff204100
+
+> kubectl create -n kube-system secret generic cilium-ipsec-keys --from-literal=keys="3 rfc4106(gcm(aes)) $PSK 128"
+secret/cilium-ipsec-keys created
+
+> kubectl describe secret -n kube-system cilium-ipsec-keys
+Name:         cilium-ipsec-keys
+Namespace:    kube-system
+Labels:       <none>
+Annotations:  <none>
+
+Type:  Opaque
+
+Data
+====
+keys:  64 bytes
+
+> SECRET="$(kubectl get secrets cilium-ipsec-keys -o jsonpath='{.data}' -n kube-system | jq -r ".keys")"
+> echo $SECRET | base64 --decode
+3 rfc4106(gcm(aes)) 47b57a0241e4c5df0a196ccfbcc7ee1aff204100 128
+
+### Start deployment with enabling encryption
+> cilium install --encryption ipsec
+
+### Check IPsec is enabled
+> cilium config view | grep ipsec
+enable-ipsec                               true
+ipsec-key-file                             /etc/ipsec/keys
+```
+
+#### Enable Multicast (W.I.P)
```

Comment:
this entire setting can be also added to cilium documentation. they have not released the feature yet, it is now staging in the main branch, expecting to be released in next release.